linera-views/src/store.rs

```diff
@@ -174,11 +174,18 @@ pub trait LocalAdminKeyValueStore: WithError + Sized {
     ) -> Result<Self, Self::Error>;
 
     /// Takes a connection and creates a new one with a different `root_key`.
-    fn clone_with_root_key(&self, root_key: &[u8]) -> Result<Self, Self::Error>;
+    async fn clone_with_root_key(&self, root_key: &[u8]) -> Result<Self, Self::Error>;
 
     /// Obtains the list of existing namespaces.
     async fn list_all(config: &Self::Config) -> Result<Vec<String>, Self::Error>;
 
+    /// Lists the root keys of the namespace.
+    /// It is possible that some root_keys have no keys.
```

Comment:
Do you mean "some namespaces have no root keys"?

Reply:
No, I really mean that.
What happens is that for each key, there is at least one root key.
But for some root key, there is no corresponding key.
* For ScyllaDb, we will have that the insertion is done via `(root_ley, key)` which lead to the fact that for each key we have a matching root_key.
* For others the root key has to be stored as a key. That creation is done just at `fn connect` or `fn clone_with_root_key`. So, the creation itself creates root keys. But the key is not yet written.